src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs

```diff
@@ -95,37 +95,52 @@ internal partial class CodeFixService : ForegroundThreadAffinitizedObject, ICode
             }
 
             using var _ = ArrayBuilder<DiagnosticData>.GetInstance(out var diagnostics);
-            using var linkedTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
-
-            var linkedToken = linkedTokenSource.Token;
 
-            // This flag is used by SuggestedActionsSource to track what solution is was
-            // last able to get "full results" for.
+            // This flag is used by SuggestedActionsSource to track what solution is was last able to get "full results" // for.
             var isFullResult = await _diagnosticService.TryAppendDiagnosticsForSpanAsync(
-                document, range, diagnostics, cancellationToken: linkedToken).ConfigureAwait(false);
+                document, range, diagnostics, cancellationToken: cancellationToken).ConfigureAwait(false);
```

Comment:
intentional switch back.  tehre is no point passing a linked token to somethign that you await prior to cancelling the token.